library/std/src/sys/windows/thread_local_key.rs

```diff
@@ -37,11 +42,14 @@ unsafe fn run_keyless_dtors() {
     // the case that this loop always terminates because we provide the
     // guarantee that a TLS key cannot be set after it is flagged for
     // destruction.
-    while let Some((ptr, dtor)) = DESTRUCTORS.pop() {
+    loop {
+        let Some((ptr, dtor)) = DESTRUCTORS.borrow_mut().pop() else {
+            break;
+        };
```

Comment:
I don't understand how this code is different?

Reply:
With a `while let` loop, the temporary is alive inside the body, which means the `RefMut` guard is already held when a destructor tries to access the list. Here, the `RefMut` is dropped immediately after the `pop`.

[demo](https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=e7df339f6cadaa2d56449199585208c3)

Reply:
Oh, that makes sense! I just wasn't thinking too clearly about the scoping rules.